modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc

```diff
@@ -0,0 +1,121 @@
+Thankfully, the `FGTimeOfDaySubsystem` offers some helpful methods for that.
+Get an instance of the `FGTimeOfDaySubsystem` with the `GetTimeSubsystem` node to make use of its `Is Day` method.
+
+However, we need to assess the time of day on multiple occasions - once when the actor spawns, and then each time it changes from day to night, or vice versa.
+For that, we can subscribe to the `OnDayStateChanged` event dispatcher of the time subsystem
+to be notified when this happens and switch our generator accordingly.
+Drag from the Return Value pin of a `Get Time Subsystem` node to create a `Bind Event to OnDayStateChanged` node.
+Next, the event (let's name it `On Day State Changed`) is created by using the `Add Custom Event...` action or by dragging from `Event` pin of `Bind Event to OnDayStateChanged` node.
+
+Hook up the execution pin of the Bind Event node to our Begin Play line since we only need to bind once.
+
+image:BeginnersGuide/simpleMod/machines/SolarPanel/EventBound.png[Binding the Event]
+
+Next, we need to set the power production depending on the passed time value.
+We'll be using the `Set Base Production` method of the power info to do this,
+since unlike the dynamic production, this power production is always "full blast" and will charge batteries.
+Use a Select Float node to select between a daytime production value and a night time production value of zero.
+We can avoid repeating outselves here and use the parent class `Get Default Power Production Capacity` method
+to retrieve the value we set in the class defaults.
+
+image:BeginnersGuide/simpleMod/machines/SolarPanel/SettingPowerProduction.png[Responding to the Event]
+
+One last thing - we need to make sure the power production is set correctly
+after a save file is loaded or the building is first created.
+To do this, call our On Day State Changed event after binding to it.
+Use the `Is Day` method to supply it with the boolean value for the time of day.
+
+image:BeginnersGuide/simpleMod/machines/SolarPanel/CallOnceOnSpawn.png[Calling on Spawn]
+
+To summarize, here's a screenshot of all of the code written in our solar panel:
+
+image:BeginnersGuide/simpleMod/machines/SolarPanel/FinishedCode.png[Finished code]
+
+That should be it, try out your newly created power generator in-game!
+
+Test it out by popping down a Power Storage and see if it is charging or not after connecting your solar panel.
+
+[TIP]
+=====
+A quick way to test the generator's time-dependent generation is to use the `SetTimeOfDay` console command.
+
+* Day time: `SetTimeOfDay 12 00`
+* Night time: `SetTimeOfDay 23 00`
+
+More info xref:SMLChatCommands.adoc[here].
+=====
+
+If you're having trouble getting your generator to work, check out the completed one in Example Mod,
+or ask for help on the Discord.
+
+== Next Steps
+
+You may wish to add functionality to make the solar panel require direct access to the sky to function,
```

Comment:
I'm guessing the naive implementation would be checking with a trace each tick. Is that so bad for performance?
A more proper implementation could be checking at startup and then only disabling when a buildable is built on top of it. Could be a good exercise to teach about "caching" as well as the buildable built event!